lib/new_relic/control/security_interface.rb

```diff
@@ -0,0 +1,40 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require 'singleton'
+
+module NewRelic
+  class Control
+    class SecurityInterface
+      include Singleton
+
+      attr_accessor :wait
+
+      def agent_started?
+        (@agent_started ||= false) == true
+      end
+
+      def waiting?
+        (@wait ||= false) == true
+      end
+
+      def init_agent
+        return if agent_started? || waiting?
+
+        if Agent.config[:'security.agent.enabled']
+          Agent.logger.info('Invoking New Relic security module')
```

Comment:
Are the security module and the security agent different things? We use both terms in the logger messages in this file.

Reply:
Kinda. I think the intention by the original author we copied and pasted from was to convey a separation between the gem / library / module which is just a chunk of Ruby code at that point, and the security agent which is the label for the running, looping application.